File: src/api/models/foodModel.js
const mongoose = require("mongoose");
const mongoosePaginate = require("mongoose-paginate");

const foodSchema = new mongoose.Schema({
  name: {
    type: String,
  },
  type: {
    type: String,
  },
  image: {
    type: String,
    default: "food_default.png",
  },
  display: {
    type: Object,
  },
  description: {
    type: String,
  },
});

foodSchema.plugin(mongoosePaginate);

module.exports = mongoose.model("Food", foodSchema);

File: client/lib/src/nutrition/models/food_model.dart
class Food {
  final dynamic id;
  final String? name;
  final String? type;
  final String? display;
  final dynamic image;
  final String? description;

  Food(
      {this.id,
      this.name,
      this.type,
      this.display,
      this.image,
      this.description});

  factory Food.fromJson(Map<String, dynamic> json) {
    return Food(
        id: json['_id'],
        name: json['name'],
        type: json['type'],
        display: json['display'],
        image: json['image'],
        description: json['description']);
  }
}
